packages/ra-core/src/i18n/translate.js

```diff
@@ -1,6 +1,7 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import wrapDisplayName from 'recompose/wrapDisplayName';
+import warning from 'warning';
```

Comment:
this package is internal to facebook (see https://github.com/facebook/fbjs), use `console.warn()` instead, and please remove usage from the code.

Reply:
> This is a mirror of Facebook's (FB) warning module used within React's source code (and other FB software). As such this module will mirror their code as much as possible.

Reply:
I remembered that I introduced this package when working on permissions. I wonder whether we should use it everywhere now. It only display warnings when not in production so that one check we don't have to verify ourselves ;)